File: 1732-find-the-highest-altitude/main.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        
        int max_altitude = 0;
        int curr_altitude = 0;
        
        for (const auto& altitude: gain) {
            curr_altitude += altitude;
            max_altitude = max(max_altitude, curr_altitude);
        }
        
        return max_altitude;
    }
};

File: 1732-find-the-highest-altitude/main.rs
impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        
        let (mut max_altitude, mut curr_altitude) = (0, 0);

        for altitude in gain.iter() {
            curr_altitude += altitude;
            max_altitude = std::cmp::max(max_altitude, curr_altitude);
        }

        max_altitude
    }
}
